evennia/objects/objects.py

```diff
@@ -1606,6 +1606,38 @@ def at_cmdset_get(self, **kwargs):
         """
         pass
 
+    def can_build_object(self):
+        """
+        This hook is called by the build command to determine if
+        self can build a new object. This is called before the
+        object is created. As it receives no arguments, it
+        can only be used to determine if self is allowed to build
+        anything in the current context.
+
+        For instance, a room may want to limit its number of exits,
+        or the builder might have an enforced quota limit for rooms
+        they can add to their custom dungeon.
+
+        Returns:
+            can_build (str or None): If self is allowed to build objects.
+                return a string as the error if there is a problem. If
+                this returns True, the build will abort.
+        """
+        pass
+
+    def at_object_constructed(self, builder):
```

Comment:
What's with these inconsistent method names :grin: 

I think the comments to [this ticket](https://github.com/evennia/evennia/issues/3095) may be a better approach here - we add a new `at_post_object_creation_hook(*args, **kwargs)` that is called differently depending on how the entity was created (such as by the spawner or by the build command). You could then have the building command pass the `caller` into it as a kwarg to do with as you want.